### PR TITLE
Polynomial Classification: Remove highcharts, other minor improvements

### PR DESCRIPTION
**Pull request by janezd**
##### Issue

Polynomial Classification uses highcharts. We'd prefer it didn't.

~~Doesn't pass tests on the oldest version of Orange, but should after merging #95, which increases the requirement.~~

##### Description of changes

Polynomial classification now uses the common `OWScatterPlotBase`.

Besides this,
- a few minor changes in GUI -- rearrangements and hiding of unnecessary controls
- removed option to set the contour step; 0.1 should be OK for everybody
- refactoring the methods

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by PrimozGodec** [comment]
It looks great 👍 👍  

I have a comment regarding colours. It seems colours (regions, dots, legend) does not match. 

Here, for example, I would expect that Iris-setosa would be coloured with blue colour. The region seems to be coloured correctly and the legend for Iris-setosa is OK, just dots seem to have the wrong colour. Also, Iris-Virginica should be green but is blue now. I like the grey background for the non-selected classes. 

Screenshot 2021-07-23 at 12 29 48

A similar problem appears when one select Iris-versicolor for a class.

**Comment by PrimozGodec** [comment]
> removed option to set the contour step; 0.1 should be OK for everybody

I think it was implemented that users can set it manually when they are too close together. For example:

Screenshot 2021-07-23 at 12 26 34

Here one would like more distributed contours. I am ok with not having the control, maybe we could think about some automation that would change the property in cases on the image. I am also OK with merging this PR with the 0.1 settings and implementing this later if we decide on automation.

File: orangecontrib/educational/widgets/owpolynomialclassification.py [hunk 1]
```diff
@@ -1,508 +1,433 @@
...
-from Orange.base import Learner as InputLearner
-from Orange.data import (
-    ContinuousVariable, Table, Domain, StringVariable, DiscreteVariable)
-from Orange.widgets import settings, gui
+import pyqtgraph as pg
+
+from Orange.data import \
+    Table, Domain, ContinuousVariable, StringVariable, DiscreteVariable
+from Orange.classification import \
+    Learner, LogisticRegressionLearner, RandomForestLearner, TreeLearner
```
Comment (PrimozGodec) [inline orangecontrib/educational/widgets/owpolynomialclassification.py:18]: Probably it is an issue that needs to be sorted out in Orange, but using `Learner` from `Orange.classification` does not allow `Tree`, `SVM` or 'Random Forest` to be connected to the widget.

**Comment by janezd** [comment]
> I have a comment regarding colours. It seems colours (regions, dots, legend) does not match.

Oops. The idea was that the target class would be colored and other class(es) would be grayed out. With this, it is easier to decypher the target class and the meaning of probabilities -- if everyhing is colored, one must look at the combo to see the target. Obviously I haven't done it properly. Now I fixed it.

One minor annoyance here: the target class is always class 1, not 0, but this looks stupid in the legend, so I had to derive a class from `OWScatterPlotBase` which reverses the legend.

File: orangecontrib/educational/widgets/owpolynomialclassification.py (hunk 1 above)
Comment (janezd) [reply orangecontrib/educational/widgets/owpolynomialclassification.py]: This is interesting. I now import from `Orange.base`, so it works.

**Comment by PrimozGodec** [comment]
> But the only effect of setting it to 0.2 is to remove some lines, or am I missing something? It does not make the plot more useful, I think. On the contrary, from dense lines one can see that the decision boundary is sharp.
> Yes, let's do so. If @BlazZupan later complains, we'll add automation (or I'll add a slider, the only sensible values are 0.1, 0.15 and 0.2, I suppose).

You are right. I didn't think about the fact that those plots with sharp edges have the same probability elsewhere (0 or 1). Agree with leaving it 0.1

**Pull request merged**